api/src/test/java/org/apache/druid/timeline/SegmentIdTest.java

```diff
@@ -0,0 +1,179 @@
+ * under the License.
+ */
+
+package org.apache.druid.timeline;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.java.util.common.Intervals;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+public class SegmentIdTest
+{
+  @Test
+  public void testBasic()
+  {
+    String datasource = "datasource";
+    SegmentId desc = SegmentId.of(datasource, Intervals.of("2015-01-02/2015-01-03"), "ver_0", 1);
+    Assert.assertEquals("datasource_2015-01-02T00:00:00.000Z_2015-01-03T00:00:00.000Z_ver_0_1", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+
+    desc = desc.withInterval(Intervals.of("2014-10-20T00:00:00Z/P1D"));
+    Assert.assertEquals("datasource_2014-10-20T00:00:00.000Z_2014-10-21T00:00:00.000Z_ver_0_1", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+
+    desc = SegmentId.of(datasource, Intervals.of("2015-01-02/2015-01-03"), "ver", 0);
+    Assert.assertEquals("datasource_2015-01-02T00:00:00.000Z_2015-01-03T00:00:00.000Z_ver", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+
+    desc = desc.withInterval(Intervals.of("2014-10-20T00:00:00Z/P1D"));
+    Assert.assertEquals("datasource_2014-10-20T00:00:00.000Z_2014-10-21T00:00:00.000Z_ver", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+  }
+
+  @Test
+  public void testDataSourceWithUnderscore1()
+  {
+    String datasource = "datasource_1";
+    SegmentId desc = SegmentId.of(datasource, Intervals.of("2015-01-02/2015-01-03"), "ver_0", 1);
+    Assert.assertEquals("datasource_1_2015-01-02T00:00:00.000Z_2015-01-03T00:00:00.000Z_ver_0_1", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+
+    desc = desc.withInterval(Intervals.of("2014-10-20T00:00:00Z/P1D"));
+    Assert.assertEquals("datasource_1_2014-10-20T00:00:00.000Z_2014-10-21T00:00:00.000Z_ver_0_1", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+
+    desc = SegmentId.of(datasource, Intervals.of("2015-01-02/2015-01-03"), "ver", 0);
+    Assert.assertEquals("datasource_1_2015-01-02T00:00:00.000Z_2015-01-03T00:00:00.000Z_ver", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+
+    desc = desc.withInterval(Intervals.of("2014-10-20T00:00:00Z/P1D"));
+    Assert.assertEquals("datasource_1_2014-10-20T00:00:00.000Z_2014-10-21T00:00:00.000Z_ver", desc.toString());
+    Assert.assertEquals(desc, SegmentId.tryParse(datasource, desc.toString()));
+  }
+
+  @Test
+  public void testDataSourceWithUnderscore2()
```

Comment:
Could you describe more about each test either in method name or in a method comment (avoiding postfixs 1, 2, ...)?

Reply:
Renamed methods and added comments.